Signup form

he-is-james opened the pull request:
uses next-input-field
incorporates Config but unsure that it works since it didn't work on my system
otherwise, pressing return on finishing a field moves to the next and pressing the submit button creates a user on the airtable form

glm1550 committed:
setup

he-is-james committed:
initial commit to track all signup fields

basic start

glm1550 committed:
housekeeping

glm1550 committed:
Merge branch 'signup-form' of https://github.com/lablueprint/apifm-mobile-app

he-is-james committed:
ignore env file

glm1550 committed:
eye icon for hiding and viewing password

he-is-james committed:
continues to next field on return

if a user is done with a field, returning will send the user to the next text input rather than manually

glm1550 committed:
merge

he-is-james committed:
update with eslint

he-is-james committed:
Merge branch 'next-input-field' into signup-form

he-is-james committed:
airtable call creates new user

error with api key, but works if hardcoded

he-is-james committed:
update yarn.lcok

he-is-james committed:
final changes for fields

added Config and adjusted fields to airtable

hwang635 committed:
Working react native config to access env variables and airtable

curesh committed:
Merge pull request #5 from lablueprint/airtableenvconfig

Working react native config to access env variables and airtable

hwang635 commented on android/app/src/main/java/com/apifm/MainApplication.java:
```diff
@@ -4,6 +4,8 @@
 import android.content.Context;
 import com.facebook.react.PackageList;
 import com.facebook.react.ReactApplication;
+import com.lugg.ReactNativeConfig.ReactNativeConfigPackage;
```
can you remove these changes / this file and also the gradlew file below

hwang635 commented on android/settings.gradle:
```diff
@@ -1,4 +1,8 @@
 rootProject.name = 'APIFM'
+include ':react-native-config'
+project(':react-native-config').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-config/android')
+include ':react-native-config'
```
duplicate lines here

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -26,6 +39,55 @@ const styles = StyleSheet.create({
+        console.log(record.getId());
+      });
+    });
+    // console.log(firstName);
```
remove these comments + other commented out console.log statements below

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -26,6 +39,55 @@ const styles = StyleSheet.create({
+      },
+    ], (err, records) => {
+      if (err) {
+        console.error(err);
```
Can you replace the console.log statements with an alert/etc that shows up on screen?

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -26,6 +39,55 @@ const styles = StyleSheet.create({
+        return;
+      }
+      records.forEach((record) => {
+        console.log(record.getId());
```
Same as above, we don't want to keep console.log statements bc they can accumulate and be confusing for devs if you don't know where they originate and they don't give any error info to the user

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -40,6 +102,77 @@ export default function SignInScreen({ navigation }) {
+          blurOnSubmit={false}
+          ref={passwordInput}
+        />
+        <Icon
```
The hide password fx is super cool

hwang635 reviewed (changes requested):
This looks really good! Please get react native config fixed if you haven't (I added steps in the env setup doc) and there are some minor edits to make that I commented on, but overall you did great work and everything looks super nice. Thanks James and Grace for your attention to detail and getting it done so fast!

he-is-james committed:
adjustments for PR

removed duplicate lines in files and finished setting up .env file, then adjusted minor edits

hwang635 reviewed (changes requested):
This looks great! One last thing -- can you add a check so that the sign up button can't be pressed/the form can't submitted unless the user has filled out all information? An alert that the user can't submit because of missing info would be great also so that they're not confused why it's not working. Right now the form can be submitted even if there's no information entered, which makes it possible to fill the airtable up with random empty rows or have a user with missing data.

glm1550 committed:
first signin screen proper fields

glm1550 committed:
checking required fields are populated

glm1550 committed:
checks required fields

glm1550 committed:
fixed order

glm1550 commented:
the form now checks if fields are populated for valid submission

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
+    else if (lastName == ""){
+      Alert.alert("Please enter the Last Name to proceed")
+    }
+    else if (email== ""){
```
The == should be "==="

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
   return (
     <View style={styles.container}>
-      <Title style={styles.titleText}> SignInScreen  :) </Title>
+      {/* <Title style={styles.titleText}> SignInScreen  :) </Title>
```
Uncomment this part or remove it

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
-          'last name': lastName,
-          'phone number': number,
-          address,
+    if (firstName == ""){
```
This looks good but I don't think it passes eslint, make sure estlint is working in your env and don't push before checking that everything passes please!

hwang635 commented on src/SignIn/SignInScreen.js:
```diff
@@ -41,46 +41,80 @@ const styles = StyleSheet.create({
-    setAddress('');
+      ], (err) => {
+        if (err) {
+          alert(err);
```
Can this be an Alert.alert like the ones above?

hwang635 reviewed (changes requested):
Hi Grace and James! Thanks for adding a submission check, everything looks pretty good. There are some minor edits to make and please pull from master to resolve merge conflicts; then it should be good to merge.

glm1550 committed:
pr requests changed

hwang635 reviewed (approved):
Beautiful thanks for working on this!
